### PR TITLE
Clear a list in a Python 2.7-compatible manner

### PR DESCRIPTION
**Pull request by hayfield**

### CODE REVIEW

File: iati/core/rulesets.py [hunk 1]
```diff
@@ -774,7 +774,7 @@ def is_valid_for(self, dataset):
                     unique.add(string_to_check)
             if len(original) != len(unique):
                 return False
-            original.clear()
+            original = list()  # Python 2 does not have a `list.clear()` function
```
Comment (allthatilk) [inline iati/core/rulesets.py]: this isn't clearing a list, it's just assigning a new empty list. A better fix would be using del or slice.

https://stackoverflow.com/questions/1400608/how-to-empty-a-list-in-python

Comment (hayfield) [reply iati/core/rulesets.py]: Since the list values aren't used on the next iteration or after the loop, it's functionally the same thing

Comment (allthatilk) [reply iati/core/rulesets.py]: it's slower and isn't as intuitive to read

**Comment by hayfield** [comment]
I'll leave this for you to merge to make sure it doesn't cause any sort of confusion with your local version of `Assumption-change-updates`

**Pull request merged**